src/composables/auth.ts

```diff
@@ -240,31 +207,97 @@ export const useAuth = createCustomScopedComposable(() => {
+    if (
+      !isAuthenticated.value
+      && isBiometricLoginEnabled.value
+      && await checkBiometricLoginAvailability()
+    ) {
+      return BiometricAuth.authenticate({
+        reason: t('biometricAuth.reason'),
+        cancelTitle: t('common.cancel'),
+        allowDeviceCredential: true,
+        iosFallbackTitle: t('biometricAuth.fallbackTitle'),
+        androidTitle: t('biometricAuth.title'),
+        androidSubtitle: t('biometricAuth.subtitle'),
+        androidConfirmationRequired: false,
+      }).then(() => {
+        isAuthenticated.value = true;
+        return true;
+      });
     }
+    return true;
   }
 
   /**
    * Open biometric login or password login modal depending on the environment settings.
+   * The modals then uses one of the `authenticateWithPassword`
+   * or `authenticateWithBiometry` methods.
    */
   async function checkUserAuth(): Promise<any> {
-    if (!isAuthenticating.value && !isAuthenticated.value) {
-      isAuthenticating.value = true;
-      if (IS_MOBILE_APP) {
-        if (isBiometricLoginEnabled.value && await checkBiometricLoginAvailability()) {
-          await openModal(MODAL_BIOMETRIC_LOGIN);
+    watchUntilTruthy(isMnemonicRestored);
+
+    if (!mnemonic.value || isAuthenticated.value) {
+      return;
+    }
+    if (isAuthenticating.value) {
+      await watchUntilTruthy(isAuthenticated);
+      return;
+    }
+
+    isAuthenticating.value = true;
+
+    if (IS_MOBILE_APP) {
+      if (isBiometricLoginEnabled.value && await checkBiometricLoginAvailability()) {
+        await openBiometricLoginModal();
+      } else {
+        isAuthenticated.value = true;
+      }
+    } else if (isMnemonicEncrypted.value) {
+      // Check if extension can be restored without requiring password
+      if (!encryptionKey.value && IS_EXTENSION && !IS_OFFSCREEN_TAB) {
+        setLoaderVisible(true);
+        const sessionEncryptionKey = await getSessionEncryptionKey();
+        if (sessionEncryptionKey) {
+          setEncryptionKey(sessionEncryptionKey);
+          mnemonicDecrypted.value = await decrypt(sessionEncryptionKey, mnemonic.value);
+          isAuthenticated.value = true;
         }
-      } else if (!encryptionKey.value) {
-        await requestUserPassword();
+        setLoaderVisible(false);
       }
 
-      // Wait before resetting isAuthenticated so that app doesn't register a false app resume event
-      await sleep(500);
+      // If restoring from background failed try to use the dev mode password set
+      // when using "Skip" when setting password.
+      if (!encryptionKey.value && UNFINISHED_FEATURES && !IS_OFFSCREEN_TAB) {
+        await authenticateWithPassword(STUB_ACCOUNT.password).catch(() => openPasswordLoginModal());
+      }
 
-      isAuthenticating.value = false;
+      // Finally if other attempts failed ask user for the password
+      if (!encryptionKey.value) {
+        await openPasswordLoginModal();
+      }
+    } else if (!isMnemonicEncrypted.value) {
```

Comment:
The mnemonic field should be migrated also for the MOBILE platform.

Reply:
For this case I created a regular migration: https://github.com/superhero-com/superhero-wallet/pull/3291/files#diff-2329098b953bf7ec6ef127a8004777a2b140d00000f0db08fc424886d85a5b46

What do you think?

Reply:
We should also remove the old mnemonic from `WalletStorage` for Mobile version.

Reply:
OK, you're right. Done. Can you check?